tools/ci_build/github/linux/docker/Dockerfile.manylinux2010_gpu

```diff
@@ -63,6 +66,36 @@ RUN CUDNN_DOWNLOAD_SUM=7eaec8039a2c30ab0bc758d303588767693def6bf49b22485a2c00bf2
     ldconfig
 
 
+# -- training dependencies --
+
+# enable training dependencies by setting build argument ENABLE_TRAINING to a non-empty value
+ARG ENABLE_TRAINING
+
+# install MPI 4.0.0
+RUN test -z $ENABLE_TRAINING || ( \
+    MPI_SUM=36f10daa3f1b1d37530f686bf7f70966b2a13c0bc6e2e05aebc7e85e3d21b10d && \
+    curl -fsSL https://download.open-mpi.org/release/open-mpi/v4.0/openmpi-4.0.0.tar.gz -O && \
+    echo "$MPI_SUM  openmpi-4.0.0.tar.gz" | sha256sum -c - && \
+    tar zxf openmpi-4.0.0.tar.gz && \
+    cd openmpi-4.0.0 && \
+    ./configure --enable-orterun-prefix-by-default && \
+    make -j $(nproc) all && \
+    make install && \
+    ldconfig \
+    )
+
+# install NCCL 2.4.8
+RUN test -z $ENABLE_TRAINING || ( \
+    RPM_SUM=ce8da101c05f0a52e01a569acc7e6845b3883f7e5760eefee85dccb279454d5d && \
+    curl -fsSL https://developer.download.nvidia.com/compute/machine-learning/repos/rhel7/x86_64/nvidia-machine-learning-repo-rhel7-1.0.0-1.x86_64.rpm -O && \
```

Comment:
Should be rhel6, not 7.
If you need 7, please use manylinux2014 docker image.

Reply:
manylinux1 is based on rhel5
manylinux2010 is based on rhel6
manylinux2014 is based on rhel7

Originally most  pieces of the code in this file was copied from https://gitlab.com/nvidia/container-images/cuda/-/blob/master/redhat/base/Dockerfile.jinja 

But recently the did some updates, it seems that it can make our life easier. 

If you have time, please take a look.  Maybe you don't have to use manylinux2014.

Reply:
added a new dockerfile for manylinux2014